### PR TITLE
Add json download to report export

### LINKED ISSUE #3366: Add Json download links next to PDF download links where they are present for Reports.
### Detailed description

We currently offer HTML and PDF views of our reports. Next to this we can also offer the JSON version of these reports. As both HTML and PDF reports are just based on those. We currently however do not allow the user to download the json anymore, but we should.

### Feature benefit/User story

As a user I want to download Json reports for safekeeping, and possibly uploading them into another organization.

### PR DESCRIPTION
**Pull request by Rieven**
### Changes
- When viewing the report we fetch the Report OOI from the id in url.
- The report OOI has the bytes id which we can fetch the report data from bytes.
- Concatenated reports needs further extraction from bytes as we have to fetch each child report data.
- Dump this data in the json export function.
- Did also some changes regarding the way templates get data, make it more consistent.

### Issue link
#3366

Closes #3366 

### QA notes

1. Test JSON download for Aggregate and Generate report  (as concatenated and single report).
2. Check if data downloaded from json matches report data from the HTML report.

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/reports/templates/partials/report_header.html [hunk 1]
```diff
@@ -13,7 +13,12 @@ <h1>{{ report_name }}</h1>
                         <li>
                             <a href="{{ report_download_pdf_url }}"
                                target="_blank"
-                               rel="noopener noreferrer"><span class="icon ti-download" aria-hidden="true"></span>{% translate "Download report" %}</a>
+                               rel="noopener noreferrer"><span class="icon ti-download" aria-hidden="true"></span>{% translate "Download report (PDF)" %}</a>
+                        </li>
+                        <li>
+                            <a href="{{ report_download_json_url }}"
+                               target="_blank"
+                               rel="noopener noreferrer"><span class="icon ti-download" aria-hidden="true"></span>{% translate "Download report (JSON)" %}</a>
```
Comment (ammar92) [inline rocky/reports/templates/partials/report_header.html:21]: This produces a duplicate `Download report` string to be translated. Both `(PDF)` and `(JSON)` strings don't have to be translated. 

E.g. `{% translate "Download report" %} (PDF)` instead

File: rocky/reports/views/base.py [hunk 2]
```diff
@@ -493,69 +493,84 @@ def get_report_data_from_bytes(self, report: ReportOOI) -> dict[str, Any]:
...
+    def get_report_data_aggregate_report(
+        self,
+    ) -> tuple[dict[str, dict[str, dict[str, Any]]], list[type[OOI]], list[dict[str, Any]]]:
+        self.bytes_client.login()
+        report_data = self.get_report_data_from_bytes(self.report_ooi)
+        children_reports = self.get_children_reports()
+        input_oois = self.get_input_oois([self.report_ooi])
+        report_types = self.get_report_types(children_reports)
+
+        return (report_data, input_oois, report_types)
```
Comment (ammar92) [inline rocky/reports/views/base.py:531]:
```suggestion
        return report_data, input_oois, report_types
```

File: rocky/reports/views/base.py [hunk 3]
```diff
@@ -493,69 +493,84 @@ def get_report_data_from_bytes(self, report: ReportOOI) -> dict[str, Any]:
...
+        report_types = self.get_report_types(children_reports)
+
+        for report in children_reports:
+            for ooi in report.input_oois:
+                report_data.setdefault(report.report_type, {})[ooi] = {
+                    "data": self.get_report_data_from_bytes(report)["report_data"],
+                    "template": report.template,
+                    "report_name": report.name,
                 }
+        return (report_data, input_oois, report_types)
```
Comment (ammar92) [inline rocky/reports/views/base.py:550]:
```suggestion
        return report_data, input_oois, report_types
```

File: rocky/rocky/locale/django.pot [hunk 4]
```diff
@@ -3494,6 +3493,14 @@ msgstr ""
 msgid "Ready"
 msgstr ""
 
+#: reports/templates/partials/report_header.html
+msgid "Download report (PDF)"
+msgstr ""
+
+#: reports/templates/partials/report_header.html
+msgid "Download report (JSON)"
+msgstr ""
```
Comment (ammar92) [inline rocky/rocky/locale/django.pot:3502]: See previous point about translations

**Review by ammar92 — COMMENTED** [review]
Tiny remarks; looks good otherwise

File: rocky/reports/templates/partials/report_header.html (hunk 1 above)
Comment (Rieven) [reply rocky/reports/templates/partials/report_header.html:21]: Good one @ammar92

**Review by ammar92 — DISMISSED** [review]

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Can download the JSON for both normal and aggregate reports, both when 1 or more reports are selected. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

File: rocky/reports/templates/partials/report_header.html [hunk 5]
```diff
@@ -13,7 +13,12 @@ <h1>{{ report_name }}</h1>
                         <li>
                             <a href="{{ report_download_pdf_url }}"
                                target="_blank"
-                               rel="noopener noreferrer"><span class="icon ti-download" aria-hidden="true"></span>{% translate "Download report" %}</a>
+                               rel="noopener noreferrer"><span class="icon ti-download" aria-hidden="true"></span>{% translate "Download report" %} (PDF)</a>
```
Comment (underdarknl) [inline rocky/reports/templates/partials/report_header.html:16]: rel="noopener noreferrer" are only needed when dealing with links to external hosts.

File: rocky/reports/views/base.py [hunk 6]
```diff
@@ -433,24 +433,16 @@ class ViewReportView(ObservedAtMixin, OrganizationView, TemplateView):
...
-
             response = {
                 "organization_code": self.organization.code,
                 "organization_name": self.organization.name,
                 "organization_tags": list(self.organization.tags.all()),
-                "data": {
-                    "report_data": {},
-                    "post_processed_data": data,
-                },
+                "data": json.loads(json.dumps(self.report_data)),
```
Comment (underdarknl) [inline rocky/reports/views/base.py:445]: Do we need this dumps and loads? They seem redundant? If the response.data is already dumped as json somewhere.

**Review by underdarknl — APPROVED** [review]

**Pull request merged**